GitUp/Application/AppDelegate.m

```diff
@@ -68,6 +79,113 @@ - (BOOL)canBecomeKeyWindow {
 
 @end
 
+@interface GIWelcomeWindow__DraggingInfoExtractor : NSObject
+- (BOOL)canDragItem:(id<NSDraggingInfo>)sender;
```

Comment:
Because there is no state in this object, I think it's perfectly acceptable to just implement these methods in the view itself. There isn't really a great benefit to having this additional object.

Reply:
@lucasderraugh 
Composition over million methods.

Reply:
My suggestion is to implement the same number of methods, with fewer classes. If this was to be reused somewhere else then sure, but as it is you're only implementing this for this one NSView subclass.